### PR TITLE
Fix two panics

### PR DESCRIPTION
**Pull request by jwilder**
* Collected would panic when it could not start and was subsequently closed.  
* Show stats would panic the DB.  Made this return an error similar to `SHOW DIAGNOSTICS`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
 ### Bugfixes
 
 - [#3180](https://github.com/influxdb/influxdb/pull/3180): Log GOMAXPROCS, version, and commit on startup.
+- [#3193](https://github.com/influxdb/influxdb/pull/3193): Fix two panics
```
Comment (otoolep) [inline CHANGELOG.md]: I suggest we enhance this note a little -- it's not much use to someone looking to see if the panic they are experiencing was fixed. Simply "Fix panic for SHOW STATS and in collectd" would be much better.

**Comment by otoolep** [comment]
+1, though I would like to see the CHANGELOG enhanced a little.

File: CHANGELOG.md (hunk 1 above)
Comment (jwilder) [inline CHANGELOG.md]: Good point. Updated.

**Comment by otoolep** [comment]
You'll want to cherry-pick, I presume.

**Comment by jwilder** [comment]
Also cherry-picked to 0.9.1.

**Pull request merged**